backend/domain_types/src/types.rs

```diff
@@ -1110,6 +1110,10 @@ impl<
         // Extract merchant_account_id from metadata before moving it
         let merchant_account_id = value.metadata.get("merchant_account_id").cloned();
 
+        // Store merchant_account_metadata for connector use
+        let merchant_account_metadata = (!value.merchant_account_metadata.is_empty())
+            .then_some(value.merchant_account_metadata.clone());
+
```

Comment:
can you pls check this if possible use this .

```suggestion
let merchant_account_metadata = (!value.merchant_account_metadata.is_empty())
    .then(|| {
        let json_value = serde_json::to_value(&value.merchant_account_metadata)
            .map_err(|_| error_stack::Report::new(ApplicationErrorResponse::InvalidData))?;
        Ok(common_utils::pii::SecretSerdeValue::new(json_value))
    })
    .transpose()?;
```